### PR TITLE
feat(TFD-146): Add tcomp jms

### PR DESCRIPTION
**Pull request by StefanLMNG**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)

There is no component available for the JMS technology.

**What is the new behavior?**

Starting point for the design-time properties of the JMS datastore, dataset and input/output components.

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
The runtime work is in progress, so the contents of the jsm-runtime_1_1 module are works in progress -- we'd really like to get a review on the jms-definition at this point and include it in the REST API to be consumed by the UI (for design-only at this point).

### CODE REVIEW

**Comment by RyanSkraba** [comment]
Hello!  We redid the PR #292 with a "clean" squashed branch.  My apologies for the first one -- it was a mistake on my part to base it off a branch in Talend/components instead of the fork!

As a result we can't see the comments and the diffs from the PR #292, but you can review the last changes from your last comments on PR #292 from the command line:

 git difftool --dir-diff pbailly/feature/TFD-146-components-jms 133091a91b23fb4b83d863bca409b13dad9f259d

**Comment by build-talend** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/602/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/602/)

File: components/components-jms/jms-definition/src/main/java/org/talend/components/jms/JmsDatasetProperties.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        return datastore;
+    }
+
+    @Override
+    public void setDatastoreProperties(JmsDatastoreProperties datastoreProperties) {
+        datastore = datastoreProperties;
+    }
+
+    @Override
+    public void setupLayout() {
```
Comment (bchen-talend) [inline components/components-jms/jms-definition/src/main/java/org/talend/components/jms/JmsDatasetProperties.java]: missing `super.setupLayout();`

File: components/components-jms/jms-definition/src/main/java/org/talend/components/jms/JmsDatastoreProperties.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+    public Property<String> httpsSettings = PropertyFactory.newString("httpsSettings");
+
+    public Property<String> property = PropertyFactory.newString("property","");
+
+    public Property<String> value = PropertyFactory.newString("value","");
+
+    @Override
+    public void setupProperties(){
+        super.setupProperties();
```
Comment (bchen-talend) [inline components/components-jms/jms-definition/src/main/java/org/talend/components/jms/JmsDatastoreProperties.java]: do you need to give version a default value here?

File: components/components-jms/jms-definition/src/main/java/org/talend/components/jms/JmsDatastoreProperties.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    public void refreshLayout(Form form) {
+        super.refreshLayout(form);
+        // Main properties
+        if (form.getName().equals(Form.MAIN)) {
+                form.getWidget(version.getName()).setVisible();
+                form.getWidget(contextProvider.getName()).setVisible();
+                form.getWidget(serverUrl.getName()).setVisible();
+                form.getWidget(userPassword.getName()).setVisible(needUserIdentity);
+                form.getWidget(useHttps.getName()).setVisible();
+                form.getWidget(httpsSettings.getName()).setVisible(useHttps);
```
Comment (bchen-talend) [inline components/components-jms/jms-definition/src/main/java/org/talend/components/jms/JmsDatastoreProperties.java]: missing afterUseHttps, and please remove all the line just with setVisible()

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/608/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/608/)

File: components/components-jms/jms-definition/src/main/java/org/talend/components/jms/input/JmsInputProperties.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+import org.talend.components.common.SchemaProperties;
+import org.talend.components.jms.JmsDatasetProperties;
+import org.talend.daikon.properties.presentation.Form;
+import org.talend.daikon.properties.property.Property;
+import org.talend.daikon.properties.property.PropertyFactory;
+
+import static org.talend.daikon.properties.property.PropertyFactory.newInteger;
+import static org.talend.daikon.properties.property.PropertyFactory.newString;
+
+public class JmsInputProperties extends ComponentPropertiesImpl {
```
Comment (bchen-talend) [inline components/components-jms/jms-definition/src/main/java/org/talend/components/jms/input/JmsInputProperties.java]: Sorry, forget to said last time, Implement IOProperties please

File: components/components-jms/jms-definition/src/main/java/org/talend/components/jms/output/JmsOutputProperties.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+import org.talend.components.jms.JmsDatasetProperties;
+import org.talend.daikon.properties.presentation.Form;
+import org.talend.daikon.properties.property.Property;
+import org.talend.daikon.properties.property.PropertyFactory;
+
+
+import static org.talend.daikon.properties.property.PropertyFactory.newBoolean;
+import static org.talend.daikon.properties.property.PropertyFactory.newEnum;
+
+public class JmsOutputProperties extends ComponentPropertiesImpl {
```
Comment (bchen-talend) [inline components/components-jms/jms-definition/src/main/java/org/talend/components/jms/output/JmsOutputProperties.java]: Same as InputProperties, implement IOProperties please

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/615/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/615/)

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/617/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/617/)

**Comment by build-talend** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/620/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/620/)

**Comment by RyanSkraba** [comment]
Hello!  We're hitting salesforce limits -- the build worked when master was merged into this branch.  It should be equivalent.

**Pull request merged**